cli/pkg/kctrl/cmd/package/release/helm_openapi_schema_gen.go

```diff
@@ -0,0 +1,350 @@
+	switch typedVal := val.(type) {
+	case *Map:
+		result := yaml2.MapSlice{}
+		typedVal.Iterate(func(k string, v interface{}) {
+			result = append(result, yaml2.MapItem{
+				Key:   k,
+				Value: h.toYAML(v),
+			})
+		})
+		return result
+	default:
+		return val
+	}
+}
+
+func (h HelmValuesSchemaGen) calculateProperties(key *yaml3.Node, value *yaml3.Node) (*Map, error) {
+	var apiKeys []*MapItem
+	description, isPresent := h.getDescriptionFromNode(key)
+	if isPresent {
+		apiKeys = append(apiKeys, &MapItem{Key: descriptionKey, Value: description})
+	}
+
+	switch value.Kind {
+	case yaml3.MappingNode:
+		var properties []*MapItem
+		apiKeys = append(apiKeys, &MapItem{Key: typeKey, Value: objectVal})
+
+		for i := 0; i < len(value.Content); i += 2 {
+			calculatedProperties, err := h.calculateProperties(value.Content[i], value.Content[i+1])
+			if err != nil {
+				return nil, err
+			}
+			properties = append(properties, &MapItem{Key: calculatedProperties.Items[0].Key, Value: calculatedProperties.Items[0].Value})
+		}
+		if len(properties) > 0 {
+			apiKeys = append(apiKeys, &MapItem{Key: propertiesKey, Value: &Map{Items: properties}})
+		} else {
+			apiKeys = append(apiKeys, &MapItem{Key: defaultKey, Value: "{}"})
+		}
+
+		sort.Sort(byKey(apiKeys))
+		if key == nil {
+			return &Map{Items: apiKeys}, nil
+		}
+		return &Map{Items: []*MapItem{&MapItem{Key: key.Value, Value: &Map{Items: apiKeys}}}}, nil
+	case yaml3.SequenceNode:
+		var defaultVals []interface{}
+		properties := &Map{[]*MapItem{}}
+		apiKeys = append(apiKeys, &MapItem{Key: typeKey, Value: arrayVal})
+
+		for _, v := range value.Content {
+			if len(v.Content) > 0 && len(v.Content[0].HeadComment) == 0 {
+				v.Content[0].HeadComment = v.HeadComment
+			}
+			switch v.Kind {
+			case yaml3.MappingNode:
+				calculatedProperties, err := h.calculateProperties(nil, v)
+				if err != nil {
+					return nil, err
+				}
+				for _, item := range calculatedProperties.Items {
+					if item.Key == propertiesKey {
+						properties.Items = append(properties.Items, item.Value.(*Map).Items...)
+					}
+				}
+			case yaml3.SequenceNode:
+				calculatedProperties, err := h.calculateProperties(nil, v)
+				if err != nil {
+					return nil, err
+				}
+				for _, item := range calculatedProperties.Items {
+					if item.Key == itemsKey {
+						properties.Items = append(properties.Items, item.Value.(*Map).Items...)
+					}
+				}
+			case yaml3.ScalarNode:
+				val, err := h.getDefaultValue(v.Tag, v.Value)
+				if err != nil {
+					return nil, err
+				}
+				defaultVals = append(defaultVals, val)
+			case yaml3.AliasNode:
+				calculatedProperties, err := h.calculateProperties(nil, v.Alias)
+				if err != nil {
+					return nil, err
+				}
+				for _, item := range calculatedProperties.Items {
+					if item.Key == itemsKey {
+						properties.Items = append(properties.Items, item.Value.(*Map).Items...)
+					}
+				}
+			default:
+				return nil, fmt.Errorf("Unrecognized type %T", v.Kind)
```

Comment:
Is there any type other than this that we do not wanna deal with?
Looks like we are doing the same thing for all types other than `ScalarNode`, could be an `if` statement

Reply:
Ideally, there is no type. I have added this as we dont know yet how to deal with if new type comes in the future.